build/ris.js

```diff
@@ -0,0 +1,397 @@
+    FERRY: {product: 'ferry', weight: 0.2},
+    SHUTTLE: {product: 'taxi', weight: 0.1}
+}
+
+const MEMBER_WEIGHT = 0.2;
+const LOWEST_PRICE_CATEGORY = 8;
+const PRICE_CATEGORY_WEIGHT = 0.5;
+
+const DAY_MAPPING = {
+    "monday": 'Mo',
+    "tuesday": 'Di',
+    "wednesday": 'Mi',
+    "thursday": 'Do',
+    "friday": 'Fr',
+    "saturday": 'Sa',
+    "sunday": 'So',
+    "holiday": 'Feiertag'
+}
+
+const abortWithError = (err) => {
+    console.error(err)
+    process.exit(1)
+}
+
+const data = new PassThrough({
+    objectMode: true,
+})
+
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+
+const exists = async (f) => {
+    try {
+        await fs.promises.stat(f);
+        return true;
+    } catch (_) {
+        return false;
+    }
+}
+
+const doRequest = (options, data) => {
+    return new Promise((resolve, reject) => {
+        const req = https.request(options, (res) => {
+            res.setEncoding('utf8');
+            let responseBody = '';
+            res.on('data', (chunk) => {
+                responseBody += chunk;
+            });
+
+            res.on('end', () => {
+                resolve(JSON.parse(responseBody));
+            });
+        });
+        req.on('error', (err) => {
+            reject(err);
+        });
+        req.write(data)
+        req.end();
+    });
+}
+
+const doRisStationsRequest = async (latitude, longitude, radius, groupBy) => {
+    const query = {
+        latitude,
+        longitude,
+        radius,
+        groupBy,
+        onlyActive: false,
+        limit: LIMIT_RESULTS
+    }
+    const options = {
+        hostname: 'apis.deutschebahn.com',
+        port: 443,
+        path: '/db-api-marketplace/apis/ris-stations/v1/stop-places/by-position?' + stringify(query),
+        method: 'GET',
+        headers: {
+            'DB-Client-Id': process.env.DB_CLIENT_ID,
+            'DB-Api-Key': process.env.DB_API_KEY,
+        },
+    };
+    console.error(options.path);
+    return doRequest(options, '');
+}
+
+const doStadaRequest = async () => {
+    const options = {
+        hostname: 'apis.deutschebahn.com',
+        port: 443,
+        path: '/db-api-marketplace/apis/station-data/v2/stations',
+        method: 'GET',
+        headers: {
+            'DB-Client-Id': process.env.DB_CLIENT_ID,
+            'DB-Api-Key': process.env.DB_API_KEY,
+            'Accept': 'application/json'
+        },
+    };
+    console.error(options.path);
+    return doRequest(options, '');
+}
+
+const geodist = (lat1, lon1, lat2, lon2) => {
+    const l1 = lat1 * Math.PI / 180, l2 = lat2 * Math.PI / 180, delta = (lon2 - lon1) * Math.PI / 180, R = 6371e3;
+    return Math.acos(Math.sin(l1) * Math.sin(l2) + Math.cos(l1) * Math.cos(l2) * Math.cos(delta)) * R;
+}
+
+const discoverRisStations = async (bbox, depth, num_requests) => {
+    if (depth > MAX_DEPTH) {
+        throw new Error('Emerg: Max depth reached');
```

Comment:
As the `bbox` & `depth`, ideally even more context, to the error to allow for debugging.

```suggestion
        const err = new Error('Emerg: Max depth reached');
        err.bbox = bbox;
        err.depth = depth;
        throw err;
```